designs/webviews/webviews-general.md

```diff
@@ -0,0 +1,95 @@
+# Webviews for Rich UIs in VS Code
+
+## Objective
+
+Justify expanding the criteria for using webviews within the AWS Toolkit for VS Code in order to create richer UIs than VS Code natively allows.
+
+### Out-of-scope
+
+* Architectures or frameworks for the webviews
+  * A best-practices guide to developing webviews with the architecture/framework decided on above.
+* Look and feel discussions
+* A detailed list of user experiences to port to webviews
+
+## Tenets (unless you know better ones)
+
+1. Users should have a delightful experience when interacting with AWS features through our AWS Toolkit.
+2. We should not limit ourselves to IDE-specific conventions if it does not provide the best user experience possible.
+3. We should do our best to not compromise the VS Code editor's or our toolkit's performance; using our toolkit should be an easy choice for an AWS developer to make, rather than a compromise.
+
+## Current State
+
+VS Code offers a very limited set of UI functionality. Specifically, VS Code offers the following constructs for user-interactible workflows:
+
+* [Quick Picks](https://code.visualstudio.com/api/extension-capabilities/common-capabilities#quick-pick), which present a list of options for a user to select. Users can select one or multiple items and can filter the list but cannot add to it.
+* [Input Boxes](https://code.visualstudio.com/api/extension-capabilities/common-capabilities#quick-pick), which allow a user to submit a string.
+* [File Pickers](https://code.visualstudio.com/api/extension-capabilities/common-capabilities#file-picker), which allow a user to choose a file or directory.
+* [Modal/Non-Modal/Status Bar Messages](https://code.visualstudio.com/api/references/vscode-api#window), which can contain arbitrary buttons.
+* [Tree Views](https://code.visualstudio.com/api/references/vscode-api#TreeView), representing hierarchical information.
```

Comment:
I assume a Tree View is a standalone thing and can't be embedded in other widgets?

Reply:
Correct. This link should be considered for the doc: https://code.visualstudio.com/api/extension-guides/tree-view